Update changelog in preparation for version 0.11.0 (#637)
probe-rs/probe-rs: CHANGELOG.md

## [Unreleased]

### Added
- Support for the `HNONSEC` bit in memory access. This now allows secure access on chips which support TrustZone (#465).
- Support for RISCV chips which use the System Bus Access method for memory access when debugging (#527).

> - Support double buffering in the flash loader (#107).

Reviewer: ```suggestion
- Support double buffering in the flash loader (#107). This improves flashing speed by about (xx%).
```

- Determine location of debug components by parsing ROM table (#431).
- Support for "flashing" data to RAM in the flash loader (#480).
- Added FTDI C232HM-DDHSL-0 to comaptible USB list for FTDI backend (#485).
- Added `--list-probes` and `-n`option to built-in GDB server binary (#486).
- Added RISCV support to GDB server (#493).
- Added `Session::target()` to access the target of a session (#497).
- Support for target description in the GDB server (#498).
- Support for register write commands in the GDB server (#510).
- Added `get_target_voltage()` function to `DebugProbe`, which can be used to read the target voltage if the probe supports it (#533).
- Added `do_chip_erase` flag to `DownloadOptions`, to allow using chip erase when flashing (#537).
- riscv: Support for memory access using system bus (#527).
- Added a generic `read` function, which can be used for memory access with maximum speed, regardless of access width (#633).
- Added an option to skip erasing the flash before programming (#628).

> - Added a new debugger for VS Code, using the [Debug Adapter Protocol](https://microsoft.github.io/debug-adapter-protocol/specification). The debugger can be found in `probe-rs-debugger` crate (#620).

Reviewer: ```suggestion
- Added a new debugger for VS Code, using the [Debug Adapter Protocol](https://microsoft.github.io/debug-adapter-protocol/specification). The debugger can be found in the `probe-rs-debugger` crate (#620).
```

- Support flashes bigger than 128 kBytes on STM32l4xx (#547).
- Added support for LPC546xx (#560).
- Added support for SiLabs EFR32 targets (#566, #567).
- Added support for flashing Intel hex files using `probe-rs-cli` (#618).
- Updated target description for NRF91 (#619).

### Changed

- Renamed `MemoryRegion::Flash` to `MemoryRegion::Nvm` (#482).
- Renamed `FlashInfo` to `NvmInfo`
- Renamed `FlashRegion` to `NvmRegion` and its `flash_info()` method to `nvm_info()`
- Renamed `FlashError::NoSuitableFlash` to `FlashError::NoSuitableNvm`
- The `into_arm_interface` and `into_riscv_interface` functions are replaced by the `try_into_arm_interface` and
  `try_into_riscv_interface` functions, which return the `Probe` struct in the case of an error. This improves the
  auto detection process (#524).
- Improved SWD protocol handling for J-Link (#443, #539, #619).
- Improved error handling for batched CMSIS-DAP commands (#445).
- Use sticky overrun behaviour for improved J-Link performance (#450).
- Better error handling for flashing (#451).
- gdb-server: Halt the chip when attaching (#461).
- Better error messages in the ram_download example (#464).

> - Cache value of CSW register (#471).

Reviewer: ```suggestion
- Cache value of CSW register to reduce used SWD line bandwidth (#471).
```

- Use `erased_byte_value` from target description as default value in the flash loader (#475).
- Added retry functionality for CMSIS-DAP probes (#462).
- riscv: Use abstract commands for CSR access for improved speed (#487).
- The `download_file` and `download_file_with_options` functions now  accept `AsRef<Path>` instead of `&Path`to be more convenient to use (#545, #579).
- Use `itm-decode` to decode ITM packets instead of built-in decoder (#564).
- Flash API Improvements: Data is now owned by the `FlashLoader`and `FlashBuilder` structs to simply the API, and the `FlashLoader::commit()` accepts the `DownloadOptions` struct instead of bool flags (#605).
- Improve internal tracking of core status (#629).
  

### Fixed

- Fixed `M33` breakpoints (#543).
- Fixed a bug where ST-Link v3 is not able to read 8 bit data chunks with more than 255 bytes. Currently we set the chunking to 128 bytes. This might be a bug in the ST-Link v3 firmware and might change in the future (#553, #609).

> - Errors occuring while trying to open J-Link probes should not prevent other probes from working (#401).

Reviewer: ```suggestion
- Errors occuring while trying to open J-Link probes do not prevent other probes from working anymore (#401).
```
